consensus/hotstuff/eventhandler/event_handler.go

```diff
@@ -114,70 +140,47 @@ func (e *EventHandler) OnReceiveProposal(proposal *model.Proposal) error {
 	log.Debug().Msg("proposal forwarded from compliance engine")
 
 	// ignore stale proposals
-	if block.View < e.forks.FinalizedView() {
+	if block.View <= e.forks.FinalizedView() {
 		log.Debug().Msg("stale proposal")
 		return nil
 	}
 
-	// we skip validation for our last own proposal
-	if proposal.Block.BlockID != e.ownProposal {
-
-		// validate the block. exit if the proposal is invalid
-		err := e.validator.ValidateProposal(proposal)
-		if model.IsInvalidBlockError(err) {
-			perr := e.voteAggregator.InvalidBlock(proposal)
-			if mempool.IsDecreasingPruningHeightError(perr) {
-				log.Warn().Err(err).Msgf("invalid block proposal, but vote aggregator has pruned this height: %v", perr)
-				return nil
-			}
-
-			if perr != nil {
-				return fmt.Errorf("vote aggregator could not process invalid block proposal %v, err %v: %w",
-					block.BlockID, err, perr)
-			}
-
-			log.Warn().Err(err).Msg("invalid block proposal")
-			return nil
-		}
+	// store the block.
+	err := e.forks.AddProposal(proposal)
+	if err != nil {
+		return fmt.Errorf("cannot add proposal to forks (%x): %w", block.BlockID, err)
+	}
 
-		if errors.Is(err, model.ErrUnverifiableBlock) {
-			log.Warn().Err(err).Msg("unverifiable block proposal")
+	_, err = e.paceMaker.ProcessQC(proposal.Block.QC)
```

Comment:
Actually I have revisited this statement and found same cases where this is possible. Your comment is correct but there is one case which is related to it. There is a case where receiving QC or TC we advance views but can't make proposal(because we miss parent proposal) in this case proposing logic needs to be triggered from `OnReceiveProposal` as well. I've made needed changes for this.

Reply:
👍 nice catch